to-do-ui/src/app/components/task-form/task-form.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit } from "@angular/core";
+import { FormBuilder, Validators } from "@angular/forms";
+import { MessageService } from "primeng/api";
+import { DynamicDialogConfig } from "primeng/dynamicdialog";
+
+import { TaskDTO } from "src/app/shared/task-dto";
+import { TaskService } from "src/app/shared/task.service";
+
+@Component({
+	selector: "app-task-form",
+	templateUrl: "./task-form.component.html",
+	styleUrls: ["./task-form.component.css"],
+	providers: [MessageService]
+})
+export class TaskFormComponent implements OnInit {
+
+	public task: TaskDTO = {};
+
+	public constructor(
+		private formBuilder: FormBuilder,
+		private messageService: MessageService,
+		private taskService: TaskService,
+		public config: DynamicDialogConfig,
+	) { }
+
+	public ngOnInit(): void {
+		this.initForm();
+	}
+
+	private initForm(): void {
+		this.taskForm.setValue({
+			title: this.config.data.title,
+			description: this.config.data.description,
+			isCompleted: this.config.data.isCompleted
+		});
+	}
+
+	public taskForm = this.formBuilder.group({
+		title: ["", Validators.required],
+		description: ["", Validators.nullValidator],
+		isCompleted: false
+	});
+
+	public onSubmit(): void {
+		const actionType = this.config.data.actionType;
+		this.createNewTaskDTO(this.taskForm.value);
+		if (actionType === "Add") {
+			this.taskService.addTask(this.task);
+		} else {
+			this.taskService.editTask(this.task);
+		}
+		this.taskForm.reset();
+	}
+
+	private createNewTaskDTO(data: Partial<any>): void { //QUESTION????
```

Comment:
```suggestion
	private createNewTaskDTO(data: Partial<any>): void { //TODO: QUESTION????
```
for highlighting
question, please!